### DISCUSSION TITLE
setError for multiple fields are not working

### DISCUSSION THREAD

**Discussion by rna** [react-hook-form/react-hook-form#10062] [Q&A]
I'm using the `setError` to generate the errors for multiple fields using forEach in the useEffect hook. But it's setting the errors only for the last field. (When I dig into it, I found that initially it's setting the errors for once for every field as required, but immediately , the change is being applied to only last field in the forEach loop.) 

```
 allCheckboxFields.forEach(fieldName => setError(fieldName, {
        type: 'manual',
        message: 'Please select and complete at least one to continue.',
      }));
```

**Comment by leapful** [comment]
@rna 

Please provide a codesandbox url to re-produce the issue.
